### PR TITLE
BUG : fix scaling issue with mlab.psd and odd NFFT

### LINKED ISSUE #4324: Inconsistency in function PSD when the NFFT parameter is an odd number
Function `mlab.psd` appears to have one inconsistency if the NFFT parameter is an odd number. In such case, as the highest frequency is not the Nyquist frequency, instead is `int(.5*NFFT)/float(dt*NFFT)`, there are two distinct values showing both at the positive and negative frequencies in the FFT result. Consider the following example:

    import numpy as np
    from matplotlib import mlab
    u = np.array([0, 1, 2, 3, 1, 2, 1])
    dt = 1.0
    Su = np.abs(np.fft.fft(u) * dt)**2 / float(dt * u.size)
    P, f = mlab.psd(u, NFFT=u.size, Fs=1/dt, window=mlab.window_none,\
        detrend=mlab.detrend_none, noverlap=0, pad_to=None, scale_by_freq=None,\
        sides='onesided')

The resulting arrays `Su` and `P` are:

    Su = [ 14.29, 1.61, 0.69, 0.55, 0.55, 0.69, 1.61 ]
    P  = [ 14.29, 3.23, 1.39, 0.55 ]

To convert `Su` to one-sided output, it is common to sum the Fourier coefficients corresponding to negative and positive frequencies. Considering that `u.size` is odd, the frequencies which correspond to the coefficients in `Su` are:

    df = 1 / float(u.size * dt)
    f = np.array([ 0, 1, 2, 3, -3, -2, -1 ]) * df

thus, as the `Su[1:4]` are the positive side coefficients and `Su[4:]` the negative side, all coefficients are summed except for `Su[0]` (which in this case, as `u` is real is the same as `Su[1:4] * 2`):

    Su_1side = np.append([Su[0]], Su[1:4] + Su[4:][::-1])

This results in `Su_1side = [ 14.29, 3.23, 1.39, 1.10 ]`, whose last coefficient differs from the result of `P`, i.e. the `mlab.psd` result.

Note that the lack of scaling of the last coefficient is not a problem when the NFFT is an even number. In such case, as the as the frequency array is:

    f = [ 0, df, 2*df, ..., (fnyquist - df), -fnyquist, -(fnyquist - df), ..., -2*df, -df ]

there is only one coefficient for the highest frequency (`fnyquist`), thus it is not summed up with another coefficient.

### PR DESCRIPTION
**Pull request by tacaswell**
closes #4324

### CODE REVIEW

**Comment by jenshnielsen** [comment]
Sensible. Should this have a test?

**Comment by tacaswell** [comment]
We don't have any other tests on mlab...

File: lib/matplotlib/mlab.py [hunk 1]
```diff
@@ -782,7 +782,17 @@ def _spectral_helper(x, y=None, NFFT=None, Fs=None, detrend_func=None,
         # Also include scaling factors for one-sided densities and dividing by
         # the sampling frequency, if desired. Scale everything, except the DC
         # component and the NFFT/2 component:
-        result[1:-1] *= scaling_factor
+
+        # if we have a even number of frequencies, don't scale NFFT/2
+        if not NFFT % 2:
+            slc = slice(1, -1, None)
+            print(NFFT)
```
Comment (jenshnielsen) [inline lib/matplotlib/mlab.py]: Leftover print?

File: lib/matplotlib/mlab.py [hunk 2]
```diff
@@ -782,7 +782,17 @@ def _spectral_helper(x, y=None, NFFT=None, Fs=None, detrend_func=None,
...
+        # if we have an odd number, just don't scale DC
+        else:
+            slc = slice(1, None, None)
+            print(NFFT)
```
Comment (jenshnielsen) [inline lib/matplotlib/mlab.py]: Same

**Comment by jenshnielsen** [comment]
Im not sure if we need a test for this but the tests in https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_mlab.py are for mlab?

**Comment by tacaswell** [comment]
Ah, I would be wrong about the tests and yes, left over prints.

**Comment by tacaswell** [comment]
@jenshnielsen Test added + prints nuked.  Thanks for keeping me honest!

**Comment by jenshnielsen** [comment]
Great. Will merge when Travis goes green

**Pull request merged**